Table in accordion

patryk0605 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

<!-- Please check all that apply using "x". -->

- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: https://github.com/mobi/goponents/issues/359


## What is the new behavior?

`go-table` supports a new binding `renderBoxShadows` and `go-accordion-panel` supports a new binding `removeContentPadding`. These two bindings when used together will allow users to have a table "flush" within a accordion.

## Does this PR introduce a breaking change?

<!-- Please check either yes or no using "x". -->

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

N/A

patryk0605 commented on projects/go-lib/src/lib/components/go-table/go-table.component.ts:
```diff
@@ -44,8 +44,8 @@ import { GoTablePage } from './go-table-page.model';
   @Input() loadingData: boolean = false;
+  @Input() renderBoxShadows: boolean = true;
   @Input() showTableActions: boolean = false;
-  @Input() stackHeader: boolean = false;
```
per discussion with @grahamhency - this binding was removed because it is believed no longer necessary. It was undocumented and not used in any examples.

grahamhency commented on projects/go-style-guide/src/app/features/ui-kit/components/accordion-docs/components/accordion-panel-docs/accordion-panel-docs.component.ts:
```diff
@@ -68,9 +68,11 @@ export class AccordionPanelDocsComponent {
+  @Input() expanded: boolean = false;
+  @Input() icon: string  = null;
+  @Input() heading: string;
+  @Input() persistState: boolean = true;
```
Thanks for adding this missing item.

grahamhency commented on projects/go-style-guide/src/app/features/ui-kit/components/table-docs/components/table-overview/table-overview.component.html:
```diff
@@ -26,6 +26,22 @@ <h4 class="go-heading-4">loadingData</h4>
+      </div>
+
+      <div class="go-column go-column--100">
+        <h4 class="go-heading-4">showTableActions</h4>
```
I was confused for a moment because I know we have documentation on this lol. Good catch on adding it here too.

patryk0605 commented:
@grahamhency I added the examples. Not sure why I didn't add them in initially 🤔. From the example for `removeContentPadding` I noticed the content was cutting into the selected green bar, so I adjusted the css for the no-padding class.
